### PR TITLE
Append model fields

### PR DESCRIPTION
**Pull request by lkraider**
This adds a method to append new fields to a model class.
Additionally, adds a copy method so that type instances can be copied.

Example:

``` python
class Person(Model):
    name = StringType()

Person.append_field('gender', Person.name.copy())
```

### CODE REVIEW

**Comment by mjrk** [comment]
+1
see https://github.com/schematics/schematics/issues/330

File: schematics/types/base.py [hunk 1]
```diff
@@ -116,6 +116,12 @@ def __init__(self, required=False, default=None, serialized_name=None,
     def __call__(self, value):
         return self.to_native(value)
 
+    def copy(self):
+        copy = object.__new__(self.__class__)
+        copy.__dict__.update(self.__dict__)
+        copy.validators = self.validators[:]
```
Comment (bintoro) [inline schematics/types/base.py]: Shouldn't `choices` and `messages` be copied as well?

**Comment by lkraider** [comment]
It's been a while since this was made, it needs updating to the latest schematics version. I am still using the 0.9.x version in production so I didn't have a need to change it yet.

**Comment by lkraider** [comment]
This is much different with Schema now, so the OP will have to create a new PR if he still wants this functionality ;)

**Pull request closed without merging**